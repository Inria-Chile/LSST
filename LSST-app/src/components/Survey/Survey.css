
h2 {
  padding: 5px 10px;
}

h3 {
  background: #2d4053;
  padding: 5px 10px;
}

.survey-container {
  width: 100%;
}

.main-skymap-wrapper {
  background: #31475c;  
  /* background: #eee; */
  /* margin: 10px; */
  padding: 5px;
  position: relative;
}

.main-container {
  position: relative;
  width: 100%;
}

.left-container {
  position: absolute;
  top: 0;
  display: inline-block;
  width: 75%;
  margin: 0px 3px;
  padding: 0px 7px;
}

.right-container {
  position: absolute;
  top: 0;
  right: 0;
  display: inline-block;
  width: 25%;
  margin: 0px 3px;
  padding: 0px 7px;
}

.minimap {
  display: inline-block;
}

.minimap-container {
  right: 0;
  top: 0;
  background: #3d5871;   
}

.mainSkymap {
  display: block;
  /* margin-left: auto;
  margin-right: auto; */
  width: 44%;
  margin: 20px auto;
}

.survey-container {
  display: inline-block;  
}

.sidebar {
    display: none;
    height: 100%;
    width: 200px;
    position: fixed;
    text-align: left;
    padding: 15px;  
    z-index: 1;
    top: 0;
    right: -200px;
    background-color: #fff;
    border-left: 1px solid #111;
    overflow-x: hidden;
    transition: 0.5s;
}

.sidebar-open {
  width: 200px;
  right: 0px;
  padding: 15px;  
}

.settings-button {
  position: absolute;
  top: 0;
  right: 0;
  margin: 10px;
}

.hover-div {
  position: absolute;
  font-size: 13px;
  bottom: 0;
  left: 0;
}

.hover-filter {
  display: inline-block;
  margin: 0px 5px;
  text-align: center;
  border-radius: 2px;
  padding: 0px 5px;
}

/* #04070a */