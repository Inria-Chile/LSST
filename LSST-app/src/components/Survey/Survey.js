import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import MainSkymap from '../Skymap/MainSkymap';
import MiniSkymaps from '../Skymap/MiniSkymaps';
import Charts from '../Charts/Charts';
import MiniScatterplot from '../Scatterplot/MiniScatterplot'
import MainScatterplot from '../Scatterplot/MainScatterplot'
import Sidebar from '../Sidebar/Sidebar';
import SurveyControls from '../SurveyControls/SurveyControls';
import ObservationsTable from '../ObservationsTable/ObservationsTable';
import './Survey.css';
import openSocket from 'socket.io-client';
import { filterColors, checkStatus, parseJSON } from "../Utils/Utils"

class Survey extends Component {
    
    static viewName = 'survey';

    constructor(props) {
        super(props);
        this.mainSkymap = null;
        this.miniSkymap = null;
        this.charts = null;
        this.miniScatterplot = null;
        this.mainScatterplot = null;
        this.displayedData = [];
        this.socket = openSocket(window.location.origin);
        this.state = {
            selectedMode: 'playback',
            selectedField: null,
            clickedField: [],
            displayedFilter: 'all',
            startDate: null,
            endDate: null,
            showSkyMap: true
        }
    }

    receiveMsg(msg){
        // console.log("received" + msg);
        msg.expDate = msg.request_time;
        this.addObservation(msg);
        this.setDate(new Date(parseInt(msg.request_time, 10)));
    }

    drawFrame = (timestamp) => {
        // console.log('Parent Drawframe')
        this.mainSkymap.drawFrame(timestamp);
        this.miniSkymap.drawFrame(timestamp);
        requestAnimationFrame(this.drawFrame);
    }
    
    setEcliptic = (show) => {
        this.mainSkymap.setEcliptic(show);
    }
    
    setGalactic = (show) => {
        this.mainSkymap.setGalactic(show);
    }
    
    setMoon = (show) => {
        this.mainSkymap.setMoon(show);
    }
    
    setTelescopeRange = (show) => {
        this.mainSkymap.setTelescopeRange(show);
    }
    
    setSidebar = (show) => {
        this.sidebar.setState({
            sidebarOpen: show
        })
    }
    
    setProjection = (proj) => {
        this.mainSkymap.setProjection(proj);
    }
    
    setDisplayedFilter = (filter) => {
        this.setState({
            displayedFilter: filter
        })
        if(this.mainSkymap!=null){
            this.mainSkymap.setDisplayedFilter(filter);
            
        }
        else{
            let showSkyMap = this.state.showSkyMap;
            this.setState({showSkyMap:!showSkyMap});
        }
       
    }

    displayScatterplot = () => {
        let showSkyMap = this.state.showSkyMap;
        this.setState({showSkyMap:!showSkyMap});
    }
    
    setDisplayedDateLimits = (startDate, endDate) => {
        if(this.state.showSkyMap)this.mainSkymap.setDisplayedDateLimits(startDate, endDate);
        this.miniSkymap.setDisplayedDateLimits(startDate, endDate);
        this.setDate(endDate);
        this.updateObservationsTable();
    }

    //function to set the start and end dates selected by the slider
    setTimeWindow = (startDate, endDate) => {
        this.startDisplayedDate = startDate;
        this.endDisplayedDate = endDate;
    }

    setLiveMode = () => {
        console.log('setlivemode')
        this.setState({
            selectedMode: 'live'
        })
        this.setData([]);
        console.log('SCOKERT', this.socket.on('data', timestamp => this.receiveMsg(timestamp)));
    }

    setPlaybackMode = () => {
        this.setState({
            selectedMode: 'playback'
        })
        this.socket.off('data');
        this.setData([]);
    }

    setDataByDate = (startDate, endDate) => {
        console.log('survey', 'Ssetdatabytdate')
        this.fetchDataByDate(startDate, endDate, (res) => {
            for(var i=0;i<res.results.length;++i)
                res.results[i]['fieldDec'] += 30;
            this.setData(res.results);
            this.setState({startDate: startDate, endDate: endDate});
            this.setDate(new Date(parseInt(endDate, 10)));
        })
    }

    setDate = (date) => {
        if(this.state.showSkyMap){
            this.mainSkymap.setDate(date);
        }
        this.miniSkymap.setDate(date);
    }
    
    fetchDataByDate = (startDate, endDate, cb) => {
        return fetch(`survey/playback/observationsCount?start_date=${startDate}&end_date=${endDate}`, {
            accept: "application/json"
        })
        .then(checkStatus)
        .then(parseJSON)
        .then(cb);
    }

    setData = (data) => {
        this.displayedData = data;
        this.charts.setData(data);
        this.miniScatterplot.setData(data);
        this.miniSkymap.setData(data);
        if(this.state.showSkyMap){
            this.mainSkymap.setData(data);
        }
        else this.mainScatterplot.setData(data);
    }

    addObservation = (obs) => {
        let added = false;
        for(let i=0;i<this.displayedData.length;++i){
            if(this.displayedData[i]['fieldID'] === obs['fieldID'] && this.displayedData[i]['filterName'] === obs['filterName']){
                // console.log('adding', obs, ' to ', this.displayedData[i]); // eslint-disable-line no-console          
                this.displayedData[i]['count']++;
                added = true;
            }
        }
        if(!added)
            this.displayedData.push(obs);
        this.setData(this.displayedData);
    }

    componentDidMount() {
        // console.log("componentDidMount")
        // this.fetchDataByDate(0, 99994323, (res) => {
        this.fetchDataByDate(0, 2, (res) => {
            for(var i=0;i<res.results.length;++i)
                res.results[i][2] += 30;
            // this.setData(res.results);
        });
    }

    cellHoverCallback = (fieldID, polygon) => {
        let latestField = null;
        let latestExpDate = 0;
        if(fieldID){
            for(let i=0;i<this.displayedData.length;++i){
                if(String(this.displayedData[i].fieldID) === String(fieldID) &&
                    (this.state.displayedFilter === this.displayedData[i].filterName || this.state.displayedFilter === 'all')){
                    if(this.displayedData[i].expDate > latestExpDate){
                        latestExpDate = this.displayedData[i].expDate;
                        latestField = this.displayedData[i];
                    }
                }
            }
        }
        this.setState({
            selectedField: latestField
        })
    }

    updateObservationsTable = () => {
        let fieldID = this.lastFieldID;
        let selectedFieldData = [];
        if(fieldID){
            for(let i=0;i<this.displayedData.length;++i){
                if(String(this.displayedData[i].fieldID) === String(fieldID) &&
                    (this.state.displayedFilter === this.displayedData[i].filterName || this.state.displayedFilter === 'all')){
                    selectedFieldData.push(this.displayedData[i]);
                }
            }
        }
        selectedFieldData.sort((a,b)=> b.expDate - a.expDate)
        this.setState({
            clickedField: selectedFieldData
        })
    }

    cellClickCallback = (fieldID, polygon) => {
        this.lastFieldID = fieldID;
        this.lastPolygon = polygon;
        this.updateObservationsTable();
        console.log('cellClickCallback');
    }

    componentDidUpdate(prevProps, prevState){
        if(prevState.displayedFilter !== this.state.displayedFilter)
            this.updateObservationsTable();
    }

    render() {
        let setters = {
            setEcliptic: this.setEcliptic,
            setGalactic: this.setGalactic,
            setMoon: this.setMoon,
            setTelescopeRange: this.setTelescopeRange,
            setSidebar: this.setSidebar,
            setProjection: this.setProjection
        }

        return (
            <div className="survey-container">
                <div>
                    <h2>
                        SURVEY PROGRESS MONITOR
                    </h2>
                    <button className="settings-button" type="button" onClick={this.setSidebar} aria-label="Settings">
                        <i className="fa fa-cog" aria-hidden="true"></i>
                    </button>
                </div>
                <div className="main-container">
                    <div className="left-container">
                        <SurveyControls setPlaybackMode={this.setPlaybackMode} 
                                        setLiveMode={this.setLiveMode} 
                                        setDataByDate={this.setDataByDate}
                                        selectedMode={this.state.selectedMode}
                                        startDate={this.state.startDate} 
                                        endDate={this.state.endDate} 
                                        setDisplayedDateLimits={this.setDisplayedDateLimits}/>
                        <Charts ref={instance => { this.charts = instance; }}/>
                        <div className="main-skymap-wrapper">
                            {!this.state.showSkyMap && <MainScatterplot ref={instance => {this.mainScatterplot=instance;}} data={this.displayedData} />}
                            {this.state.showSkyMap && <MainSkymap ref={instance => { this.mainSkymap = instance; }} 
                                data={this.displayedData} 
                                filter={this.state.displayedFilter}
                                startDate = {this.state.startDate} 
                                endDate={this.state.endDate}
                                cellHoverCallback={this.cellHoverCallback} 
                                cellClickCallback={this.cellClickCallback} />}
                            {
                                this.state.selectedField &&
                                <div className="hover-div">
                                    <div>FieldID: {this.state.selectedField && this.state.selectedField.fieldID ? this.state.selectedField.fieldID: ''}</div>
                                    <div>Timestamp: {this.state.selectedField && this.state.selectedField.expDate ? this.state.selectedField.expDate: ''}</div>
                                    <div>
                                        Filter: {" "}
                                        <div className="hover-filter" style={{
                                                backgroundColor: filterColors[this.state.selectedField.filterName ] ? filterColors[this.state.selectedField.filterName ]
                                                    : "#000000",
                                            }}>
                                            {
                                                this.state.selectedField.filterName ? this.state.selectedField.filterName : ''
                                            }
                                        </div>
                                    </div>
                                </div>
                            } 
                        </div>
                        <div>
                            <ObservationsTable selectedField={this.state.selectedField} clickedField={this.state.clickedField} />
                        </div>
                    </div>
                    <div className="right-container">
                        <MiniSkymaps ref={instance => { this.miniSkymap = instance; }} onMinimapClick={this.setDisplayedFilter} />
                        <MiniScatterplot ref={instance => {this.miniScatterplot=instance;}} onScatterplotClick={this.displayScatterplot}/>
                    </div>
                </div>
                <Sidebar ref={instance => { this.sidebar = instance; }} {...setters} skymap={this.mainSkymap} />
            </div>
        );
    }
}

export default Survey;
