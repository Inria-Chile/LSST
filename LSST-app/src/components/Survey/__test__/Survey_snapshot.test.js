import React from 'react';
import Survey from '../Survey';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
global.fetch = require('jest-fetch-mock');

describe('Survey snapshot test',function(){
    let res = {
        results : []
    }       
    const firstElement = {airmass: 1.3828,
        count: 1,
        expDate: 756893245000,
        expTime: 34,
        fieldDec: -40.227999999999994,
        fieldID: 166,
        fieldRA: 91.217,
        filterName: "z",
        lst: 1.0884,
        rotSkyPos: 0.375884,
        seeing: 1.00583,
        skybrightness_modified: 19.4018}
    
    const secondElement = {
        airmass: 1.3828,
        count: 1,
        expDate:  756894245000,
        expTime: 34,
        fieldDec: -40.227999999999996,
        fieldID: 166,
        fieldRA: 91.217,
        filterName: "z",
        lst: 1.0884,
        rotSkyPos: 0.375884,
        seeing: 1.00583,
        skybrightness_modified: 19.4018
    }
    res.results.push(firstElement);
    res.results.push(secondElement);
    fetch.mockResponse(JSON.stringify(res));

    it('renders correctly',()=>{
        const tree = shallow(<Survey/>);
        expect(toJson(tree)).toMatchSnapshot();
        console.log(tree.state().hoveredField);
    });

    it('shows cellHoverInfo when a field in the skymap is selected',()=>{
        const selectedField = {
            airmass: 2.49149,
            count: 1,
            expDate: 2594813,
            expTime: 34,
            fieldDec: -50.2,
            fieldID: 39,
            fieldRA: 196.858,
            filterName: "y",
            lst: 1.24181,
            rotSkyPos: 2.30015,
            seeing: 0.98908,
            skybrightness_modified: 17,  
        };
        const tree = shallow(<Survey/>);
        tree.setState({hoveredField :selectedField})
        expect(toJson(tree)).toMatchSnapshot();      
    });

    it('shows scatterplot instead of mainskymap when scatterplot container is clicked',()=>{  
        const tree = shallow(<Survey/>);
        let scatterplotContainer = tree.find('.scatterplot-container');
        scatterplotContainer.simulate('click');
        expect(toJson(tree)).toMatchSnapshot();    
    });

    it('shows mainskymap when scatterplot container is clicked',()=>{
        const tree = shallow(<Survey/>);
        let scatterplotContainer = tree.find('.scatterplot-container');
        scatterplotContainer.simulate('click');
        scatterplotContainer.simulate('click');
        expect(toJson(tree)).toMatchSnapshot();    
    });


});