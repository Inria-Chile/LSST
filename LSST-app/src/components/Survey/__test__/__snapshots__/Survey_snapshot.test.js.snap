// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey snapshot test renders correctly 1`] = `
<div
  className="survey-container"
>
  <DraggableTitle
    title="Survey Progress"
  />
  <div
    className="main-container"
  >
    <div
      className="left-container"
    >
      <SurveyControls
        endDate={null}
        selectedMode="playback"
        setDataByDate={[Function]}
        setDisplayedDateLimits={[Function]}
        setLiveMode={[Function]}
        setPlaybackMode={[Function]}
        setTimeWindow={[Function]}
        startDate={null}
      />
      <div
        className="bottom-left-container"
      >
        <Charts
          mode="playback"
        />
        <div
          className="row"
        >
          <div
            className="col-6"
          >
            <div
              className="main-skymap-wrapper"
            >
              <Settings
                setEcliptic={[Function]}
                setGalactic={[Function]}
                setMoon={[Function]}
                setProjection={[Function]}
                setTelescopeRange={[Function]}
              />
              <div
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <Skymap
                  cellClickCallback={[Function]}
                  cellHoverCallback={[Function]}
                  cellUpdateCallback={[Function]}
                  className="mainSkymap"
                  displayedData={Array []}
                  filter="all"
                  nodeRef="mainNode"
                  projection="aitoff"
                  showEcliptic={true}
                  showGalactic={true}
                  showMoon={true}
                  showTelescopeRange={true}
                />
              </div>
              <div
                style={
                  Object {
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="scatterplot"
                >
                  <Scatterplot
                    displayedData={Array []}
                    height={400}
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6"
          >
            <ObservationsTable
              clickedField={Array []}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="right-container"
    >
      <MiniSkymaps
        displayedData={Array []}
        fontSize={0}
        gridOpacity={0}
        onMinimapClick={[Function]}
        selectedFilter="all"
      />
      <div
        className="scatterplot-container"
        onClick={[Function]}
      >
        <h4>
           Right Ascention / Declination 
        </h4>
        <Scatterplot
          displayedData={Array []}
          height={183}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Survey snapshot test shows cellHoverInfo when a field in the skymap is selected 1`] = `
<div
  className="survey-container"
>
  <DraggableTitle
    title="Survey Progress"
  />
  <div
    className="main-container"
  >
    <div
      className="left-container"
    >
      <SurveyControls
        endDate={null}
        selectedMode="playback"
        setDataByDate={[Function]}
        setDisplayedDateLimits={[Function]}
        setLiveMode={[Function]}
        setPlaybackMode={[Function]}
        setTimeWindow={[Function]}
        startDate={null}
      />
      <div
        className="bottom-left-container"
      >
        <Charts
          mode="playback"
        />
        <div
          className="row"
        >
          <div
            className="col-6"
          >
            <div
              className="main-skymap-wrapper"
            >
              <Settings
                setEcliptic={[Function]}
                setGalactic={[Function]}
                setMoon={[Function]}
                setProjection={[Function]}
                setTelescopeRange={[Function]}
              />
              <div
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <Skymap
                  cellClickCallback={[Function]}
                  cellHoverCallback={[Function]}
                  cellUpdateCallback={[Function]}
                  className="mainSkymap"
                  displayedData={Array []}
                  filter="all"
                  nodeRef="mainNode"
                  projection="aitoff"
                  showEcliptic={true}
                  showGalactic={true}
                  showMoon={true}
                  showTelescopeRange={true}
                />
              </div>
              <div
                style={
                  Object {
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="scatterplot"
                >
                  <Scatterplot
                    displayedData={Array []}
                    height={400}
                  />
                </div>
              </div>
              <CellHoverInfo
                selectedField={
                  Object {
                    "airmass": 2.49149,
                    "count": 1,
                    "expDate": 2594813,
                    "expTime": 34,
                    "fieldDec": -50.2,
                    "fieldID": 39,
                    "fieldRA": 196.858,
                    "filterName": "y",
                    "lst": 1.24181,
                    "rotSkyPos": 2.30015,
                    "seeing": 0.98908,
                    "skybrightness_modified": 17,
                  }
                }
              />
            </div>
          </div>
          <div
            className="col-6"
          >
            <ObservationsTable
              clickedField={Array []}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="right-container"
    >
      <MiniSkymaps
        displayedData={Array []}
        fontSize={0}
        gridOpacity={0}
        onMinimapClick={[Function]}
        selectedFilter="all"
      />
      <div
        className="scatterplot-container"
        onClick={[Function]}
      >
        <h4>
           Right Ascention / Declination 
        </h4>
        <Scatterplot
          displayedData={Array []}
          height={183}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Survey snapshot test shows mainskymap when scatterplot container is clicked 1`] = `
<div
  className="survey-container"
>
  <DraggableTitle
    title="Survey Progress"
  />
  <div
    className="main-container"
  >
    <div
      className="left-container"
    >
      <SurveyControls
        endDate={null}
        selectedMode="playback"
        setDataByDate={[Function]}
        setDisplayedDateLimits={[Function]}
        setLiveMode={[Function]}
        setPlaybackMode={[Function]}
        setTimeWindow={[Function]}
        startDate={null}
      />
      <div
        className="bottom-left-container"
      >
        <Charts
          mode="playback"
        />
        <div
          className="row"
        >
          <div
            className="col-6"
          >
            <div
              className="main-skymap-wrapper"
            >
              <Settings
                setEcliptic={[Function]}
                setGalactic={[Function]}
                setMoon={[Function]}
                setProjection={[Function]}
                setTelescopeRange={[Function]}
              />
              <div
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <Skymap
                  cellClickCallback={[Function]}
                  cellHoverCallback={[Function]}
                  cellUpdateCallback={[Function]}
                  className="mainSkymap"
                  displayedData={Array []}
                  filter="all"
                  nodeRef="mainNode"
                  projection="aitoff"
                  showEcliptic={true}
                  showGalactic={true}
                  showMoon={true}
                  showTelescopeRange={true}
                />
              </div>
              <div
                style={
                  Object {
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "width": "100%",
                  }
                }
              >
                <div
                  className="scatterplot"
                >
                  <Scatterplot
                    displayedData={Array []}
                    height={400}
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6"
          >
            <ObservationsTable
              clickedField={Array []}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="right-container"
    >
      <MiniSkymaps
        displayedData={Array []}
        fontSize={0}
        gridOpacity={0}
        onMinimapClick={[Function]}
        selectedFilter="all"
      />
      <div
        className="scatterplot-container"
        onClick={[Function]}
      >
        <h4>
           Right Ascention / Declination 
        </h4>
        <Scatterplot
          displayedData={Array []}
          height={183}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Survey snapshot test shows scatterplot instead of mainskymap when scatterplot container is clicked 1`] = `
<div
  className="survey-container"
>
  <DraggableTitle
    title="Survey Progress"
  />
  <div
    className="main-container"
  >
    <div
      className="left-container"
    >
      <SurveyControls
        endDate={null}
        selectedMode="playback"
        setDataByDate={[Function]}
        setDisplayedDateLimits={[Function]}
        setLiveMode={[Function]}
        setPlaybackMode={[Function]}
        setTimeWindow={[Function]}
        startDate={null}
      />
      <div
        className="bottom-left-container"
      >
        <Charts
          mode="playback"
        />
        <div
          className="row"
        >
          <div
            className="col-6"
          >
            <div
              className="main-skymap-wrapper"
            >
              <Settings
                setEcliptic={[Function]}
                setGalactic={[Function]}
                setMoon={[Function]}
                setProjection={[Function]}
                setTelescopeRange={[Function]}
              />
              <div
                style={
                  Object {
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "width": "100%",
                  }
                }
              >
                <Skymap
                  cellClickCallback={[Function]}
                  cellHoverCallback={[Function]}
                  cellUpdateCallback={[Function]}
                  className="mainSkymap"
                  displayedData={Array []}
                  filter="all"
                  nodeRef="mainNode"
                  projection="aitoff"
                  showEcliptic={true}
                  showGalactic={true}
                  showMoon={true}
                  showTelescopeRange={true}
                />
              </div>
              <div
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <div
                  className="scatterplot"
                >
                  <Scatterplot
                    displayedData={Array []}
                    height={400}
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6"
          >
            <ObservationsTable
              clickedField={Array []}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="right-container"
    >
      <MiniSkymaps
        displayedData={Array []}
        fontSize={0}
        gridOpacity={0}
        onMinimapClick={[Function]}
        selectedFilter="all"
      />
      <div
        className="scatterplot-container"
        onClick={[Function]}
      >
        <h4>
           Right Ascention / Declination 
        </h4>
        <Scatterplot
          displayedData={Array []}
          height={183}
        />
      </div>
    </div>
  </div>
</div>
`;
